draft-barnes-acme.md

```diff
@@ -1603,87 +1604,90 @@ failed.
-   name in the SNI field, the server presents the generated certificate.
+The client will generate a self-signed certificate with the subject's
+organizationName field set to the "signature" value from the JWS, i.e., the
+base64-encoded signature value and a subjectAlternativeName extension containing
+a single dNSName of "\<token\>.acme.invalid".  The client will then configure
+the TLS server at the domain such that when a handshake is initiated with the
+Server Name Indication extension set to "\<token\>.acme.invalid", the generated
+test certificate is presented.
 
-The client's response provides its random value S:
+The response to the DVSNI challenge provides the validation JWS to the server.
 
 type (required, string):
 : The string "dvsni"
 
-s (required, string):
-: A random 32-byte secret octet string, Base64-encoded
+validation (required, string):
+: Base64-encoded SHA-256 hash of the DER-encoded challenge certificate
 
 ~~~~~~~~~~
 {
   "type": "dvsni",
-  "s": "9dbjsl3gTAtOnEtKFEmhS6Mj-ajNjDcOmRkp3Lfzm3c"
+  "validation": {
+    "header": { "alg": "HS256" },
+    "payload": "qzu9...6bjn",
+    "signature": "gfj9XqFv07e1wU66hSLYkiFqYakPSjAu8TsyXRg85nM"
+  }
 }
 ~~~~~~~~~~
 
 Given a Challenge/Response pair, the ACME server verifies the client's control
-of the domain by verifying that the TLS server was configured as expected:
-
-1. Compute the value Z = SHA-256(R &#124;&#124; S)
-2. Open a TLS connection to the domain name being validated on port 443,
-   presenting the value "\<nonce\>.acme.invalid" in the SNI field.
-3. Verify the following properties of the certificate provided by the TLS
-server:
-  * It is a valid self-signed certificate
-  * The public key is the public key for the key pair being authorized
-  * It contains the domain name being validated as a subjectAltName
-  * It contains a subjectAltName matching the hex-encoding of Z, with the suffix
-    ".acme.invalid"
+of the domain by verifying that the TLS server was configured appropriately.
+
+1. Verify the validation JWS using the account key for which the challenge was
+   issued
+2. Decode the payload of the JWS as UTF-8 encoded JSON
+3. Verfiy that there are exactly two fields in the decoded object, and that:
+  * The "type" field is set to "dvsni"
+  * The "token" field matches the "token" value in the challenge
+4. Open a TLS connection to the domain name being validated on port 443,
+   presenting the value "\<token\>.acme.invalid" in the SNI field.
+5. Verify the following properties of the certificate provided by the TLS server:
+  * The organizationName field in the subject matches the "signature" value in
+    the "validation" JWS
+  * The certificate contains a single subjectAltName of the form
+    "\<token\>.acme.invalid".
 
 It is RECOMMENDED that the ACME server verify the challenge certificate using
 multi-path probing techniques to reduce the risk of DNS hijacking attacks.
```

Comment:
This line should also be changed.  You don't "verify the certificate with multiple paths", you "perform validation TLS connections from multiple vantage points".